### PR TITLE
 Inital draft for Pred docs

### PR DESCRIPTION
**Pull request by galileopy**
Documentation for the Pred crock, with a nice example :)

### CODE REVIEW

File: docs/crocks/Pred.md [hunk 1]
```diff
@@ -2,12 +2,65 @@
 
 `Pred a`
 
---
+The `Pred` crock allows you to reuse the same predicate logic with different kind of inputs, the only think we ask of you is that you let `Pred` know how to get from your expected input to the predicate input.
```
Comment (evilsoft) [inline docs/crocks/Pred.md]: > the only think

should be only thing

File: docs/crocks/Pred.md [hunk 2]
```diff
@@ -2,12 +2,65 @@
...
+
+Say you have a predicate function `isOfAge = age => age >= 21`, but you have an input `{name: "John", age : 23}`, you could of course do something like `isPersonOfAge = person => isOfAge(person.age)`
+
+Or you can use `Pred` and gain a monoidal interface, with the ability to decorate your predicate to work with any kind of input, using the contramap interface.
+
+That means that you would have the ability to easily chain your predicates without too much rewrite.
+
+Let's say that you also want to require that a person was referred by a friend, you could just concat 2 preds or use another mechanism for conjunction.
```
Comment (evilsoft) [inline docs/crocks/Pred.md]: > 2 preds

should be:
(2) `Pred`s

File: docs/crocks/Pred.md [hunk 3]
```diff
@@ -2,12 +2,65 @@
 
 `Pred a`
 
---
+The `Pred` crock allows you to reuse the same predicate logic with different kind of inputs, the only think we ask of you is that you let `Pred` know how to get from your expected input to the predicate input.
+
+Say you have a predicate function `isOfAge = age => age >= 21`, but you have an input `{name: "John", age : 23}`, you could of course do something like `isPersonOfAge = person => isOfAge(person.age)`
+
+Or you can use `Pred` and gain a monoidal interface, with the ability to decorate your predicate to work with any kind of input, using the contramap interface.
```
Comment (evilsoft) [inline docs/crocks/Pred.md]: > using the contramap

should be:
using the `contramap`

File: docs/crocks/Pred.md [hunk 4]
```diff
@@ -2,12 +2,65 @@
...
+
+That means that you would have the ability to easily chain your predicates without too much rewrite.
+
+Let's say that you also want to require that a person was referred by a friend, you could just concat 2 preds or use another mechanism for conjunction.
 
---
 
 ```js
---
+// Helpers -- predicate functions
```
Comment (evilsoft) [inline docs/crocks/Pred.md]: With these examples we should show the inclusion of the functions into the "file" as plucked off of `crocks` in alphabetical order like so:

```js
const { 
  Pred, compose, concat, contramap,
  curry, option, prop
} = crocks
```

Just so they can get a feel of what is provided.

File: docs/crocks/Pred.md [hunk 5]
```diff
@@ -2,12 +2,65 @@
...
+const person =
```
Comment (evilsoft) [inline docs/crocks/Pred.md]: Would prefer (2) space indentation. and a line break between declarations. Like this:

```js
const person =
  {name: "John", age : 23, referral: "ANON"}

const isOfAge =
  age => age >= 21

const isReferred =
  who => who === "FRIEND"
```

File: docs/crocks/Pred.md (hunk 1 above)
Comment (rstegg) [inline docs/crocks/Pred.md]: After this sentence can you add a sentence to let the user know their return type will be coerced to a boolean?

Comment (rstegg) [inline docs/crocks/Pred.md]: After this sentence can you add a sentence to let the user know their return type will be coerced to a boolean?

**Comment by evilsoft** [comment]
Closing this for now. Going to go a different route with dat documentation.

**Pull request closed without merging**